services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java

```diff
@@ -346,14 +348,20 @@ public void getStreamSchemas(CommandGetSchemas commandGetSchemas,
   public FlightInfo getFlightInfoTables(CommandGetTables commandGetTables,
                                         CallContext callContext,
                                         FlightDescriptor flightDescriptor) {
-    throw CallStatus.UNIMPLEMENTED.withDescription("CommandGetTables not supported.").toRuntimeException();
+    final Schema schema = getSchemaTables().getSchema();
+
+    return getFlightInfoForFlightSqlCommands(commandGetTables, flightDescriptor, schema);
   }
 
   @Override
   public void getStreamTables(CommandGetTables commandGetTables,
                               CallContext callContext, Ticket ticket,
                               ServerStreamListener serverStreamListener) {
-    throw CallStatus.UNIMPLEMENTED.withDescription("CommandGetTables not supported.").toRuntimeException();
+    final CallHeaders headers = retrieveHeadersFromCallContext(callContext);
```

Comment:
See comments on the CommandGetSchemas PR about some refactoring work.

Reply:
Fixed